### PR TITLE
Cleaning up client request unit tests

### PR DESCRIPTION
**Pull request by krousey**
1. Closing test servers that we open.
2. Remove dependency on Config and just call NewRESTClient directly
3. Remove auth tests as they are redundant to the tests in transport_test.go

@kubernetes/goog-csi 
@kubernetes/sig-api-machinery

### CODE REVIEW

File: pkg/client/unversioned/request_test.go [hunk 1]
```diff
@@ -959,9 +962,6 @@ func TestDoRequestNewWayFile(t *testing.T) {
 	requestURL := testapi.Default.ResourcePathWithPrefix("foo/bar/baz", "", "", "")
 	requestURL += "?timeout=1s"
 	fakeHandler.ValidateRequest(t, requestURL, "POST", &tmpStr)
-	if fakeHandler.RequestReceived.Header["Authorization"] == nil {
```
Comment (caesarxuchao) [inline pkg/client/unversioned/request_test.go]: There are other occurrences in this file (e.g., L864). Why not removing them altogether?

**Comment by caesarxuchao** [comment]
A nit, otherwise LGTM. Thanks.

File: pkg/client/unversioned/request_test.go (hunk 1 above)
Comment (krousey) [inline pkg/client/unversioned/request_test.go]: The answer to that is, I missed a lot on my search. I will make sure I get them all this time.

Comment (krousey) [inline pkg/client/unversioned/request_test.go]: Thanks for the catch.

**Comment by krousey** [comment]
Should have caught all the instances of it now. Thanks.

Also I changed a benchmark variable from `t` to `b` as is the custom.

**Comment by caesarxuchao** [comment]
LGTM. Thanks.

**Comment by krousey** [comment]
I only squashed. Readding LGTM.

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Pull request merged**